### ISSUE TITLE
jax.nn.log_softmax loses small finite float64 gradient component

### ISSUE THREAD

**Issue by ALinrunrun** [jax-ml/jax#40384]
### Description

`jax.nn.log_softmax` appears to return an incorrect zero value for one component of a first-order `float64` gradient.

`log_softmax` is invariant to a common shift of all logits, so the input VJP should sum to zero. In the example below, the expected gradient is approximately:

```
[5.551115123125776e-17, -5.551115123125776e-17]
```

Both components are finite, normal, nonzero `float64` values. However, JAX returns `0.0` for the first component, while preserving the second component.

The same issue is also relevant to the mapped public path `jax.scipy.special.log_softmax`.

### Minimal reproducible example

```
#!/usr/bin/env python3
import os
import sys
import math
import numpy as np

os.environ.setdefault("CUDA_VISIBLE_DEVICES", "-1")
os.environ.setdefault("JAX_PLATFORMS", "cpu")
os.environ.setdefault("TF_CPP_MIN_LOG_LEVEL", "3")

try:
    import jax
    import jax.numpy as jnp
except ImportError as e:
    print(f"missing dep: {e}")
    sys.exit(2)

jax.config.update("jax_enable_x64", True)

z = jnp.asarray([37.42994775023705, 0.0], dtype=jnp.float64)
cotangent = jnp.asarray([1.0, 0.0], dtype=jnp.float64)

expected = np.asarray(
    [5.551115123125776e-17, -5.551115123125776e-17],
    dtype=np.float64,
)

actual = np.asarray(
    jax.grad(lambda value: jnp.vdot(jax.nn.log_softmax(value, axis=-1), cotangent))(z),
    dtype=np.float64,
)

print("jax:", actual.tolist())
print("expected:", expected.tolist())
print("VJP sum:", float(np.sum(actual)))

if not np.allclose(actual, expected, rtol=1e-6, atol=1e-12, equal_nan=True):
    sys.exit(0)
sys.exit(1)
```

### Actual result

```
jax: [0.0, -5.551115123125776e-17]
expected: [5.551115123125776e-17, -5.551115123125776e-17]
VJP sum: -5.551115123125776e-17
```

### Expected result

```
[5.551115123125776e-17, -5.551115123125776e-17]
```

The expected gradient components are finite, normal, nonzero, and representable in `float64`. JAX instead drops the first component to zero, and the resulting VJP no longer preserves the zero-sum property expected from common-shift invariance.

### System info (python version, jaxlib version, accelerator, etc.)

jax: 0.10.1
jaxlib: 0.10.1
numpy: 2.4.6
python: 3.13.13
platform: Ubuntu 24.04.3 LTS, Linux x86_64
accelerator: CPU backend

Note: GPU is present, but CUDA-enabled jaxlib is not installed, so JAX falls back to CPU.

**Comment by jakevdp** [comment]
This falls under #38651 – issues like this (typically discovered via AI fuzzing of extreme input values) are low priority for the JAX team.
